src/uct/ib/rdmacm/rdmacm_ep.c

```diff
@@ -61,14 +61,12 @@ static UCS_CLASS_INIT_FUNC(uct_rdmacm_ep_t, uct_iface_t *tl_iface,
          * to proceed with the connection establishment.
          * This event will be retrieved from the event_channel by the async thread.
          * All endpoints share the interface's event_channel but can use it serially. */
-         if (rdma_resolve_addr(iface->cm_id, NULL, (struct sockaddr *)sockaddr->addr,
-                               UCS_MSEC_PER_SEC * iface->config.addr_resolve_timeout)) {
-             ucs_error("rdma_resolve_addr(addr=%s) failed: %m",
-                       ucs_sockaddr_str((struct sockaddr *)sockaddr->addr,
-                                        ip_str, ip_len));
-             status = UCS_ERR_IO_ERROR;
-             goto err_free_mem;
-         }
+        status = uct_rdmacm_resolve_addr(iface->cm_id, (struct sockaddr *)sockaddr->addr,
```

Comment:
i would add another wrapper function - uct_rdmacm_ep_resolve_addr(ep) - and use it here and from uct_rdmacm_iface_client_start_next_ep(), since all parameters are the same and need to pass only ep. only need to set self->remote_addr up front
NOTE: I've noticed that `self->remote_addr = (struct sockaddr *)(sockaddr->addr)` is a bug because the user may release the pointer it passed to this function. need to hold a copy inside rdmacm_ep (sockaddr_in/6 or sockaddr_storage)

Reply:
@yosefe you mean to add a second function that would call rdma_resolve_addr?
wouldn't it be better to call if from a single place like you suggested in the previous CR comment? 
also, uct_rdmacm_iface_client_start_next_ep takes the parameters from the iface, not the ep, and the current ep doesnt have a pointer to its iface. 
i wanted to make the ucx function that calls the rdmacm api function, accept the same parameters, it would also prevent the derefernce inside this function on every call.

Reply:
@alinask i meant the 2nd function will call uct_rdmacm_resolve_addr.
also, every ep always has a pointer to the iface, in super.super ..

Reply:
@yosefe yes, i meant a direct pointer to the iface. adding this wrapper is adding an additional step (function call) before calling rdma_resolve_addr on the remote address. not sure i see its benefit but since it's not the fast path i guess it's not so bad.